### PR TITLE
Add documentation, organize requests files, add collection requests and data utils

### PR DESCRIPTION
**Pull request by Richar-p**
- Init YARD to create documentation
- Organize requests files
- Add collections requests
- Add DataUtils module
- Fix Specs
- Fix CI

### CODE REVIEW

File: lib/kaze_client/data_utils.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+module KazeClient
+  # @author richar_p@modulotech.fr
+  # Utility methods to interact with KazeData
+  # @since 0.4.0
+  module DataUtils
+    # Parse a Kaze Workflow datetime
+    # @param time [Integer] The datetime to parse
+    # @return [Time] The parsed datetime
+    # @example
+    #  KazeClient::KazeWorkflowUtils.parse_datetime(1546300800000)
```
Comment (Ezveus) [inline lib/kaze_client/data_utils.rb]:
```suggestion
    #  KazeClient::DataUtils.parse_datetime(1546300800000)
```

File: lib/kaze_client/data_utils.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # => 2019-01-01 00:00:00 UTC
+    def self.parse_time(time)
+      Time.at(time.to_i / 1000).utc
```
Comment (Ezveus) [inline lib/kaze_client/data_utils.rb:17]: You should handle when time is `nil` and probably when `time` is a `String` too.

File: lib/kaze_client/request/request.rb [hunk 3]
```diff
@@ -116,14 +116,10 @@ def non_generic_error(error, message, response)
...
       case response.code
-      when 401
-        KazeClient::Error::Unauthorized.new(message)
-      when 403
-        KazeClient::Error::Forbidden.new(message)
-      when 404
-        KazeClient::Error::NotFound.new
-      when 500
-        KazeClient::Error::InternalServerError.new(message)
+      when 401 then KazeClient::Error::Unauthorized.new(message)
```
Comment (Ezveus) [inline lib/kaze_client/request/request.rb:119]: Why this change?

File: lib/kaze_client/request/requests/collections/items/items_request.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  # @author richar_p@modulotech.fr
+  # List the items assigned to the collection id.
+  # @see KazeClient::Request
+  # @see KazeClient::Utils::FinalRequest
+  # @see KazeClient::Utils::AuthentifiedRequest
+  # @since 0.4.0
+  class ItemsRequest < Utils::FinalRequest
+    include Utils::AuthentifiedRequest
+    include Utils::ListRequest
+    def initialize(id)
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/collections/items/items_request.rb:14]:
```suggestion

    def initialize(id)
```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module KazeClient
+  # @author RICHARD Peter <richar_p@modulotech.fr>
+  # KazeClient request to add tags to a job
+  # @note    This request is for the moment in re-development. It's not working well
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
  # @note    This request is for the moment in re-development. It is not working well.
```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  #          When you add a tag with it, it doesnt check if the id is correct or not and it delete all the other tags
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
  #          When you add a tag with it, it does not check if the id is correct or not and it deletes all the other tags.
```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  #          It will be fixed in the next version of Kaze API, but for now, we use it like that
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
  #          It should be fixed in a next version of Kaze API.
```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  #   {
+  #       "tags": [1, 2, 3]
+  #   }
+  # @return [KazeClient::Response] The response of the request
+  # @since 0.4.0
+  class AddTagRequest < Utils::FinalRequest
+    include Utils::AuthentifiedRequest
+
+    def initialize(job_id, tag_ids)
+      super(:put, "/api/tags/references/job/#{job_id}.json")
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
      super(:put, "/api/tags/references/job/#{job_id}")

```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  #          It will be fixed in the next version of Kaze API, but for now, we use it like that
+  #
+  # @example
+  #     rq = KazeClient::AddTagRequest.new(123, [1, 2, 3]).with_token('token')
+  #     KazeClient::Client.new('https://kaze.modulotech.fr').execute(rq)
+  # Will make a request like this:
+  #    PUT https://kaze.modulotech.fr/api/tags/references/job/123.json
+  #   --- HEADERS ---
+  #   {
+  #       "Authorization": "token"
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
  #       "Authorization": "token", "Content-Type": "application/json", "Accept": "application/json"
```

File: lib/kaze_client/request/requests/jobs/add_tag_request.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  # KazeClient request to add tags to a job
+  # @note    This request is for the moment in re-development. It's not working well
+  #          When you add a tag with it, it doesnt check if the id is correct or not and it delete all the other tags
+  #          It will be fixed in the next version of Kaze API, but for now, we use it like that
+  #
+  # @example
+  #     rq = KazeClient::AddTagRequest.new(123, [1, 2, 3]).with_token('token')
+  #     KazeClient::Client.new('https://kaze.modulotech.fr').execute(rq)
+  # Will make a request like this:
+  #    PUT https://kaze.modulotech.fr/api/tags/references/job/123.json
```
Comment (Ezveus) [inline lib/kaze_client/request/requests/jobs/add_tag_request.rb]:
```suggestion
  #    PUT https://kaze.modulotech.fr/api/tags/references/job/123
```

**Review by Ezveus — CHANGES_REQUESTED** [review]

File: lib/kaze_client/data_utils.rb (hunk 2 above)
Comment (Richar-p) [reply lib/kaze_client/data_utils.rb:17]: Agree for the nil

But for the string, i use .to_i to handle it.

File: lib/kaze_client/request/request.rb (hunk 3 above)
Comment (Richar-p) [reply lib/kaze_client/request/request.rb:119]: To be rubocop compliant

**Comment by Richar-p** [comment]
Up! :)

**Comment by Ezveus** [comment]
Manually merged.

**Pull request closed without merging**